subxt/src/utils.rs

```diff
@@ -96,3 +96,137 @@ unsafe impl<T> Sync for PhantomDataSendSync<T> {}
 /// with collections like BTreeMap. This has the same type params
 /// as `BTreeMap` which allows us to easily swap the two during codegen.
 pub type KeyedVec<K, V> = Vec<(K, V)>;
+
+/// Generic `scale_bits` over `bitvec`-like `BitOrder` and `BitFormat` types.
+pub mod bits {
+    use codec::{
+        Compact,
+        Input,
+    };
+    use scale_bits::{
+        scale::format::{
+            Format,
+            OrderFormat,
+            StoreFormat,
+        },
+        Bits,
+    };
+    use std::marker::PhantomData;
+
+    macro_rules! store {
+        ($ident: ident; $(($ty: ident, $wrapped: ty)),*) => {
+            /// Associates `bitvec::store::BitStore` trait with corresponding, type-erased `scale_bits::StoreFormat` enum.
+            ///
+            /// Used to decode bit sequences in runtime by providing `scale_bits::StoreFormat` using
+            /// `bitvec`-like type type parameters.
+            pub trait $ident {
+                /// Corresponding `scale_bits::StoreFormat` value.
+                const FORMAT: StoreFormat;
+                /// Number of bits that the backing store types holds.
+                const BITS: u32;
+            }
+
+            $(
+                impl $ident for $wrapped {
+                    const FORMAT: StoreFormat = StoreFormat::$ty;
+                    const BITS: u32 = <$wrapped>::BITS;
+                }
+            )*
+        };
+    }
+
+    macro_rules! order {
+        ($ident: ident; $($ty: ident),*) => {
+            /// Associates `bitvec::order::BitOrder` trait with corresponding, type-erased `scale_bits::OrderFormat` enum.
+            ///
+            /// Used to decode bit sequences in runtime by providing `scale_bits::OrderFormat` using
+            /// `bitvec`-like type type parameters.
+            pub trait $ident {
+                /// Corresponding `scale_bits::OrderFormat` value.
+                const FORMAT: OrderFormat;
+            }
+
+            $(
+                #[doc = concat!("Type-level value that corresponds to `scale_bits::OrderFormat::", stringify!($ty), "` at run-time")]
+                #[doc = concat!(" and `bitvec::order::BitOrder::", stringify!($ty), "` at the type level.")]
+                #[derive(Clone, Debug, PartialEq, Eq)]
+                pub enum $ty {}
+                impl $ident for $ty {
+                    const FORMAT: OrderFormat = OrderFormat::$ty;
+                }
+            )*
+        };
+    }
+
+    store!(BitStore; (U8, u8), (U16, u16), (U32, u32), (U64, u64));
+    order!(BitOrder; Lsb0, Msb0);
+
+    /// Constructs a run-time format parameters based on the corresponding type-level parameters.
+    pub fn bit_format<Store: BitStore, Order: BitOrder>() -> Format {
+        Format {
+            order: Order::FORMAT,
+            store: Store::FORMAT,
+        }
+    }
```

Comment:
does this need to be `pub`? (I don't think we need to expose it anywhere?)

Reply:
We certainly don't need to but if we're exposing bridge types for both `bitvec` and `scale-bits`, I thought it doesn't hurt to expose this convenience function as well. Should I hide it?

Reply:
I guess I just can't think of a use for it offhand; people can encode the decoded bits again back into the same format, or they can iterate over the Bits and encode to any other format, or what have you :)

Plus, I tend to lean towards keeping the API surface small I suppose just to keep our options open!

Reply:
Sure, makes sense! Done in https://github.com/paritytech/subxt/pull/718/commits/9bdcd73ac129bcf86f3b5363c9821c8ec2826700